nservicebus/azure-service-bus/configuration/configuration.md

```diff
@@ -0,0 +1,200 @@
+ * `NamespacePartitioning()`: Provides access to the settings that determine how entities are partitioned across namespaces.
+ * `Composition()`: Provides access to the settings that determine how entities are composed inside a single namespace.
+ 
+### Validation
+
+These settings determine how entity names are validated. These settings should correspond to the actual validation rules in place in the Azure Service Bus namespace configured. Different types of namespaces have different implementations and these implementations have changed over time as well (without notice and sometimes confusing/outdated msdn documentation). The default settings align with recently created Standard namespaces.
+
+ * `UseQueuePathMaximumLength(int)`: The maximum length of a queue path (path = name + namespace hierarchy), defaults to 260.
+ * `UseTopicPathMaximumLength(int)`: The maximum length of a topic path (path = name + namespace hierarchy), defaults to 260.
+ * `UseSubscriptionPathMaximumLength(int)`: The maximum length of a subscription path (path = name), defaults to 50.
+ * `UseStrategy<T>()`: An implementation of `IValidationStrategy` that validates the entity path. Following implementations exist:
+	 * `EntityNameValidationV6Rules`: allows letters, numbers, periods (.), hyphens (-), and underscores (-)
+	 * `EntityNameValidationRules` (default): allows letters, numbers, periods (.), hyphens (-), underscores (-) and slashes (/) for queues and topics, no slashes allowed for subscriptions.
+
+### Sanitization
+
+Sanitization refers to the cleanup logic, that turns invalid entity names into valid ones.
+
+ * `UseStrategy<T>()`: An implementation of `ISanitizationStrategy` that handles invalid entity names. The following implementations exist:
+	 * `AdjustmentSanitizationV6`: removes invalid characters according to `EntityNameValidationV6Rules`, uses MD5 hashing to reduce the length of an entity name if the maximum length is exceeded.
+	 * `AdjustmentSanitization` (default): removes invalid characters according to `EntityNameValidationRules`, uses SHA1 hashing to reduce the length of an entity name if the maximum length is exceeded.
+	 * `ThrowOnFailingSanitization`: throws an `EndpointValidationException` if the name is invalid.
+
+### Individualization
+
+Individualization refers to the logic that modifies an entity name so that the entity can be differentiated (or not) between different instances of the same endpoint.
+
+ * `UseStrategy<T>()`: An implementation of `IIndividualizationStrategy` that modifies an endpoint name to become unique per endpoint instance. Following implementations exist:
+	 * `CoreIndividualization` (default): Makes no modifications, and relies on the individualization logic as defined in the NServiceBus core framework.
+	 * `DiscriminatorBasedIndividualization`: modifies the name of the endpoint by appending a discriminator value to the end.
+ 
```

Comment:
I see. This is only there for backward compat, so this should be added to the upgrade guide and not here, as going forward individualization will be done by the core.